tests/test_data.py

```diff
@@ -0,0 +1,187 @@
+from json import dumps
+import pytest
+from time import sleep
+
+from data import DataFactory, DataV2, DataV3
+import topics
+
+def test_data_factory():
```

Comment:
Happy with how the tests are done now. Next PR, can you please group and split up unit tests using classes. https://docs.pytest.org/en/latest/getting-started.html#group-multiple-tests-in-a-class

Reply:
May as well do it now while I'm making changes to the tests

Reply:
I do like having them split up! I didn't even realise it was an option